channeldb/channel_test.go

```diff
@@ -1531,3 +1534,77 @@ func TestFinalHtlcs(t *testing.T) {
 	_, err = cdb.LookupFinalHtlc(chanID, unknownHtlcID)
 	require.ErrorIs(t, err, ErrHtlcUnknown)
 }
+
+// TestSerializeHTlCs tests serialization of HTLCs combined with extra data.
+func TestSerializeHTlCs(t *testing.T) {
+	t.Parallel()
+
+	mockHtlc := HTLC{
+		Signature:     testSig.Serialize(),
+		Incoming:      false,
+		Amt:           10,
+		RHash:         key,
+		RefundTimeout: 1,
+		OnionBlob:     lnmock.MockOnion(),
+		ExtraData:     make([]byte, 0),
+	}
+
+	testCases := []struct {
+		name  string
+		htlcs []HTLC
+	}{
+		{
+			// Serialize multiple HLTCs with no extra data to
+			// assert that there is no regresion for HTLCs with
+			// no extra data.
+			name: "no extra data",
```

Comment:
Not sure what you mean here? If we fail on HTLCs with no extra data, we'd fail for all legacy HTLCs?

Reply:
I meant a new case to test the error `ErrOnionBlobLength` is returned as expected when conditions were met.

Reply:
Ah, right! That's difficult to add in this test because we can't make `SerializeHtlcs` write an incorrect onion blob length so got to have something that does the incorrect serialization manually. 

I think this is _sorta_ fragile to have a unit test that mimics our encoding (if we change `SerializeHtlcs` we have to separately change the unit test) _but_ it's a nice peace of mind test to have - will add on next push!

Reply:
although also, if we're changing `SerializeHtlcs` means we're doing a megaa migration / already in a world of pain and one pesky unit test is the least of our problems :')